File: python/minimize-maximum-pair-sum-in-array.py
class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        min_pair_sum = 0
        nums.sort()
        for i in range(len(nums) // 2):
            j = len(nums) - i - 1
            min_pair_sum = max(min_pair_sum, nums[i] + nums[j])
        return min_pair_sum

File: go/minimize-maximum-pair-sum-in-array.go
import (
    "sort"
)

func minPairSum(nums []int) int {
    minPair := 0
    sort.Ints(nums)
    for i := 0; i < len(nums) / 2; i++ {
        candidate := nums[i] + nums[len(nums) - i - 1]
        if minPair < candidate {
            minPair = candidate
        }
    }
    return minPair
}
